coalib/misc/Annotations.py

```
@@ -0,0 +1,44 @@
+
+
+def typechain(*args):
+    """
+    Returns function which applies the first transformation it can from args
+    and returns transformed value, raises ValueError when cannot apply none.
+
+    >>> function = typechain(int, ord, None)
+    >>> function("10")
+    10
+    >>> function("a")
+    97
+    >>> function(int)
+    <class 'int'>
+    >>> function(None) is None
+    True
+    >>> function("str")
+    Traceback (most recent call last):
+        ...
+    ValueError: Couldn't convert to any specified type value 'str'.
+
+    :raises TypeError:  Raises when either no functions are specified for
+                        checking or one or more object isn't callable and not
+                        None.
+    """
+    if len(args) == 0:
+        raise TypeError("No arguments were provided.")
```

well it's just a little bit more generic if you allow no arguments, it's less code that achieves the same thing ultimately. (Although I'm also a fan of fail fast)